signer/src/lib.rs

```diff
@@ -1231,16 +1231,17 @@ pub fn verify_voucher_signature(
     }
 }
 
-/// Return the CID of a signed message
+/// Return the CID of an unsigned message
 ///
 /// # Arguments
 ///
-/// * `signed_message_api` - The signed message;
-pub fn get_cid(signed_message_api: SignedMessageAPI) -> Result<String, SignerError> {
-    let signed_message = SignedMessage::try_from(&signed_message_api)?;
-    let cbor_signed_message = to_vec(&signed_message)?;
+/// * `unsigned_message_api` - The unsigned message;
+pub fn get_cid(unsigned_message_api: UnsignedMessageAPI) -> Result<String, SignerError> {
```

Comment:
Here we need to support both signed and unsigned message. It should be possible by testing `MessageTxAPI` to see if it match one or the other type.

Reply:
I added the support for `MessageTxAPI` in the wasm-bindgen part, and from there it just gets the underlying `UnsignedMessageAPI`, would you like me to move the handling for that over here?

Reply:
Yes because we can use also the crate on its own.
Also I missed your question in the comment before. You can get 2 kinds of CID from unsigned message and from signed message. That's why here we need to be able to do both cases. Before I was only doing signed message because it this CID that is used to track the transaction state but we should also be able to give the CID of an unsigned message if someone need it.

You have some info here on what Filecoin CID is https://spec.filecoin.io/#section-libraries.multiformats

Reply:
Is it wrong to assume that `SignedMessage.cid()` (importing `Cbor`) and the procedure you had manually should return the same CID?
Because right now a test for me is failing by relying on `.cid()` so I wanted to know which one I should use...

Reply:
So they should be the same for the signed message. It should not give a different CID unless I have made a mistake in our crafted CID in the previous implementation.

Reply:
Yeah it seems the there's a mismatch between the previous implementation and the one with the trait, for `SignedMessage`